src/theory/quantifiers/skolemize.cpp

```diff
@@ -100,30 +104,27 @@ TrustNode Skolemize::process(Node q)
   return TrustNode::mkTrustLemma(lem, pg);
 }
 
-bool Skolemize::getSkolemConstants(Node q, std::vector<Node>& skolems)
+std::vector<Node> Skolemize::getSkolemConstants(const Node& q)
 {
-  std::unordered_map<Node, std::vector<Node>>::iterator it =
-      d_skolem_constants.find(q);
-  if (it != d_skolem_constants.end())
+  Assert(q.getKind() == EXISTS);
+  std::vector<Node> skolems;
+  for (size_t i = 0, nvars = q[0].getNumChildren(); i < nvars; i++)
   {
-    skolems.insert(skolems.end(), it->second.begin(), it->second.end());
-    return true;
+    skolems.push_back(getSkolemConstant(q, i));
```

Comment:
If I understand correctly, generating the Skolems like this will create applications of the QUANTIFIERS_SKOLEMIZE function with arguments `q` and `i` for the i-th var. This does not directly relate, say, `QUANTIFIERS_SKOLEMIZE(q, 1)` to `QUANTIFIERS_SKOLEMIZE(q, 0)`, while the former is defined in terms of the latter. Is that right? So it would be the job of whoever is using the skolems to reconstruct this relationship?

Reply:
Yes, this is analogous to how other inferences that introduce skolems work.

Reply:
I see. This will boil down to, for example, the Alethe node converter have to implement what getwitnessform used to do.